src/lib/accessibility/FocusHandling/FocusContainer.ts

```diff
@@ -0,0 +1,75 @@
+import { IFocusableItemProps, SetStateDelegate } from './FocusableItem'
+
+export interface IFocusContainerProps<T> {
+  items?: T[]
+}
+
+export interface IFocusContainerState {
+  focusItemOnIdx: number
+}
+
+export class ContainerFocusHandler<
+  T,
+  P extends IFocusContainerProps<T>,
+  S extends IFocusContainerState
+> {
+  constructor(
+    private getProps: () => P,
+    private setState: SetStateDelegate<P, S>,
+    private initState: (state: IFocusContainerState) => void,
+    private getState: () => S,
+  ) {
+    this.initState({ focusItemOnIdx: 0 })
+  }
+
+  public assignAtomicItemsProps(idx: number, itemsLength: number): IFocusableItemProps {
+    const itemProps: IFocusableItemProps = {
+      isFocused: idx === this.getState().focusItemOnIdx && this.getState().focusItemOnIdx !== -1,
+      isFirstElement: idx === 0,
+      isLastElement: idx === itemsLength - 1,
+      onMovePrevious: this.movePrevious.bind(this),
+      onMoveNext: this.moveNext.bind(this),
+      onMoveFirst: this.moveFirst.bind(this),
+      onMoveLast: this.moveLast.bind(this),
+      onEnter: this.enter.bind(this),
+      onSpace: this.space.bind(this),
+      onEsc: this.esc.bind(this),
+    }
+
+    return itemProps
+  }
+
+  private movePrevious(): void {
+    this.setState(prev => {
+      return { focusItemOnIdx: prev.focusItemOnIdx - 1 }
```

Comment:
there are two ways of doing this: 
 - add a loop, which will verify the item state
 - wrap items into smart LinkedList, which will handle this automatically
but before doing this, we have to introduce component state (the state which reflects if the component is disabled/active/focused etc). This will be done as a separate PR and after we can proceed with it